### PR TITLE
Add storageClass info as http header during GET API call

### PR DESCRIPTION
**Pull request by prateekpandey14**
Signed-off-by: prateekpandey14 <prateekpandey14@gmail.com>

### CODE REVIEW

File: openebs/pkg/volume/v1alpha1/volume.go [hunk 1]
```diff
@@ -133,7 +133,7 @@ func (v CASVolume) CreateVolume(vol v1alpha1.CASVolume) error {
 }
 
 // ReadVolume to get the info of CAS volume through a API call to m-apiserver
-func (v CASVolume) ReadVolume(vname string, namespace string, obj interface{}) error {
+func (v CASVolume) ReadVolume(vname string, namespace string, storageclass string, obj interface{}) error {
```
Comment (princerachit) [inline openebs/pkg/volume/v1alpha1/volume.go]: how about `name,namespace,storageclass string` ?

File: openebs/pkg/volume/v1alpha1/volume.go [hunk 2]
```diff
@@ -150,6 +150,7 @@ func (v CASVolume) ReadVolume(vname string, namespace string, obj interface{}) e
 	}
 
 	req.Header.Set("namespace", namespace)
+	req.Header.Set(string(v1alpha1.StorageClassKey), storageclass)
```
Comment (princerachit) [inline openebs/pkg/volume/v1alpha1/volume.go]: Some comments would help here.

File: openebs/pkg/volume/v1alpha1/volume.go (hunk 1 above)
Comment (prateekpandey14) [reply openebs/pkg/volume/v1alpha1/volume.go]: works , i will change.

File: openebs/pkg/volume/v1alpha1/volume.go (hunk 2 above)
Comment (prateekpandey14) [reply openebs/pkg/volume/v1alpha1/volume.go]: done!

File: openebs/pkg/volume/v1alpha1/volume.go [hunk 3]
```diff
@@ -150,6 +150,9 @@ func (v CASVolume) ReadVolume(vname string, namespace string, obj interface{}) e
 	}
 
 	req.Header.Set("namespace", namespace)
+	// passing storageclass info as a request header which will extracted by the
+	// Maya-apiserver to get the CAS template name
+	req.Header.Set(string(v1alpha1.StorageClassKey), storageclass)
```
Comment (AmitKumarDas) [inline openebs/pkg/volume/v1alpha1/volume.go:155]: There has been a change in maya api server w.r.t storage class key. Please refer to PR https://github.com/openebs/maya/pull/414

**Review by AmitKumarDas — APPROVED** [review]

**Review by AmitKumarDas — APPROVED** [review]

**Review by princerachit — APPROVED** [review]
lgtm

**Pull request merged**